developer/src/kmc-model/test/test-compile-trie.ts

```diff
@@ -160,5 +160,15 @@ describe('createTrieDataStructure()', function () {
     })
     assert.match(uppercaseSourceCode, /"key":\s*"TURTLES"/);
     assert.notMatch(uppercaseSourceCode, /"key":\s*"turtles"/);
+  });
+
+  it('does not create `null`/"undefined"-keyed children', function () {
+    const WORDLIST_FILENAME = makePathToFixture('example.qaa.wordlist-ordering', 'wordlist.tsv');
+    // check to ensure the Trie is fully well-formed.
+    let sourceCode = createTrieDataStructure([WORDLIST_FILENAME], (wf) => wf);
+    assert.notMatch(sourceCode, /"undefined"/);
+    assert.notMatch(sourceCode, /null/);
```

Comment:
This adequately tests the effect of the bug but it doesn't really test the actual validity of the outcome very well. I wonder if there is a more rigorous test we could apply?

Reply:
The obvious "most rigorous" approach would be to spin up a full `TrieModel` - or maybe just its 'traversal' - and ensure that all wordlist entries are indeed accessible.  That'd add a test dependency on `@keymanapp/models-templates`, though.  I can do that if we're fine with making that link.

Reply:
I think that a full TrieModel sounds like a good idea, but perhaps it should go in a separate PR, particularly as it adds an extra dependency? I am comfortable with this one merging as-is.

Reply:
Sounds like a plan:  see #11085.